compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -1925,6 +1925,26 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     *impl_,
                 );
                 if let Some(assoc_ty) = assoc_ty {
+                    let (assoc_ident, def_scope) =
+                        tcx.adjust_ident_and_get_scope(assoc_ident, adt_def.did(), hir_ref_id);
+
+                    let kind = DefKind::AssocTy;
+                    if !assoc_ty.visibility(tcx).is_accessible_from(def_scope, tcx) {
+                        let kind = kind.descr(assoc_ty.def_id);
+                        let msg = format!("{kind} `{assoc_ident}` is private");
+                        let def_span = self
+                            .tcx()
+                            .hir()
+                            .span_if_local(assoc_ty.def_id)
+                            .unwrap_or_else(|| self.tcx().def_span(assoc_ty.def_id));
+                        tcx.sess
+                            .struct_span_err_with_code(span, &msg, rustc_errors::error_code!(E0624))
+                            .span_label(span, &format!("private {kind}"))
+                            .span_label(def_span, &format!("private {kind} defined here"))
```

Comment:
```suggestion
                            .span_label(def_span, &format!("{kind} defined here"))
```
It may not have been defined as private, it may have been defined as `pub(super)` but still out of `def_scope`'s reach.